src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,43 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
+import java.lang.reflect.InvocationTargetException;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
```

Comment:
The `getAll` method should ideally return a list of the specific type `T` rather than `Machine`. Consider changing the method signature to `public List<T> getAll(Class<? extends T> type)` if the class is parameterized with `T`.